src/interfaces/wallet.cpp

```diff
@@ -60,16 +60,16 @@ WalletTx MakeWalletTx(CWallet& wallet, const CWalletTx& wtx)
 }
 
 //! Construct wallet tx status struct.
-WalletTxStatus MakeWalletTxStatus(interfaces::Chain::Lock& locked_chain, const CWalletTx& wtx)
+WalletTxStatus MakeWalletTxStatus(CWallet& wallet, const CWalletTx& wtx)
 {
     WalletTxStatus result;
     result.block_height = wtx.m_confirm.block_height > 0 ? wtx.m_confirm.block_height : std::numeric_limits<int>::max();
     result.blocks_to_maturity = wtx.GetBlocksToMaturity();
     result.depth_in_main_chain = wtx.GetDepthInMainChain();
     result.time_received = wtx.nTimeReceived;
     result.lock_time = wtx.tx->nLockTime;
-    result.is_final = locked_chain.checkFinalTx(*wtx.tx);
-    result.is_trusted = wtx.IsTrusted(locked_chain);
+    result.is_final = wallet.chain().checkFinalTx(*wtx.tx);
```

Comment:
what is the risk of this getting out-of-sync with `result.block_height` or other members? 9229faa752

Reply:
re: https://github.com/bitcoin/bitcoin/pull/16426#discussion_r417376070

> what is the risk of this getting out-of-sync with `result.block_height` or other members like `result.is_trusted`? [9229faa](https://github.com/bitcoin/bitcoin/commit/9229faa752cdd7bc778a67a638effb273ad7de19)

I think only risk is gui briefly displaying transaction as final in the transaction HTML description before wallet processing catches up and other fields in the HTML description reflect that. This is probably a minor concern but #17443 should fix it

Reply:
Oh, I wasn't aware this is only used for the gui